### ISSUE TITLE
Problem getting js to load on one page

### ISSUE THREAD

**Issue by Gmfholley** [Gmfholley/07-21-Restful-Routes-With-Rails#2]
I am trying to use ajax for my posts class.  

I have put my script into the posts.js file that Rails creates for me in the app/assets/javascripts folder. I am trying to add some event listeners to elements on a single view.  I am adding them using a window.onload function.  

To test the window.onload function, my very first line is an alert.  I have tried window.onload using this code.

```js
  my_onload_function() {
    alert("It worked!");
    // some other stuff
  }
  window.addEventListener("load", my_onload_function, false); 
```

And also this code:

```js
window.onload = function() {
  alert("It loaded!");
  // some other stuff
}
```

What I expect to happen is that any time I load any posts view, that the alert will fire.

What actually happens is that it sometimes fires and sometimes does not.  If I reload the page, it fires every time.

Since I only need this javascript for a single view, I have also attempted to add this file directly to the file.  I put it in the bottom of my view file with this code: 
```html-erb
<script type="text/javascript" src="/app/assets/javascripts/posts.js"></script>
```

And I have tried a few other relative paths.  When I try to do this, I get a Rails error saying there are no routes matching that path. 

I think there is something in rails that only loads my js file once, so it's not loading on the pages it should.  Is there a way to bypass this mechanism or a way to add a js file to just a single view?

**Comment by sumeetjain** [comment]
> I think there is something in rails that only loads my js file once, so it's not loading on the pages it should. Is there a way to bypass this mechanism or a way to add a js file to just a single view?

Yes. In the stretches for the project description (https://github.com/omahacodeschool/class-drakon/issues/89), I mention disabling turbolinks for a couple of the JS-related stretches. If you do that, you should be good to go.
